chsdi/templates/htmlpopup/gletschermaechtigkeit_gpr_profiles.mako

```diff
@@ -5,6 +5,6 @@
   layer = c['layerBodId']
 %>
 <tr><td class="cell-left">${_(layer + '.gpr_prf_name')}</td> <td>${c['attributes']['gpr_prf_name'] or '-'}</td></tr>
-<tr><td class="cell-left">${_(layer + '.gpr_max_thik')}</td> <td>${c['attributes']['gpr_max_thik'] or '-'}</td></tr>
+<tr><td class="cell-left">${_(layer + '.gpr_max_thik')}</td> <td>${int(c['attributes']['gpr_max_thik']) or '-'}</td></tr>
```

Comment:
@faselm it is not such a good idea to round the numbers here. It is better to make sure they are already integers in the DB or even better that they are integers in the delivered data? If I remember correct there was already a discussion on this on the first integration of the layer: https://github.com/geoadmin/mf-chsdi3/pull/3585

Reply:
good point, we should ask the data-owner for a delivery with rounded value, that would be the most correct way and we're not manipulating / changing any data, just publishing.

if thats not possible, please do the rounding with the fme import, then we have the same data in identify requests and htmlpopup requests or create a view which is doing the type cast.

Reply:
It's not rounding the numbers, it's only to remove the decimals. I will change the data types of the columns, though

Reply:
thank you.  just wanted to be sure that we publish the same data here in the [identify](https://mf-chsdi3.dev.bgdi.ch/ltfas_swisstopo_geologie_gletschermaechtigkeit/rest/services/all/MapServer/identify?geometry=2585152.4609374995,1083800.2929687498&geometryFormat=geojson&geometryType=esriGeometryPoint&imageDisplay=1250,948,96&lang=de&layers=all:ch.swisstopo.geologie-gletschermaechtigkeit&limit=10&mapExtent=2580926.250765216,1080945.3375428233,2587176.250765216,1085685.3375428233&returnGeometry=true&sr=2056&tolerance=10) and in the [htmlpopup](https://mf-chsdi3.dev.bgdi.ch/ltfas_swisstopo_geologie_gletschermaechtigkeit/2103152022/rest/services/ech/MapServer/ch.swisstopo.geologie-gletschermaechtigkeit/8151/htmlPopup?coord=2585152.4609374995,1083800.2929687498&imageDisplay=1250,948,96&lang=de&mapExtent=2580926.250765216,1080945.3375428233,2587176.250765216,1085685.3375428233&sr=2056).

It seems to be ok for mean_thick and max_thick but not for volume.

Reply:
Hi, it seems the volumes are identical in both links above (58.29). The volume will be in million m3 instead of m3 in the new version..

Reply:
there are not the same values:
[identify / db](https://mf-chsdi3.dev.bgdi.ch/ltfas_swisstopo_geologie_gletschermaechtigkeit/rest/services/all/MapServer/identify?geometry=2585231.3574218745,1083917.0214843748&geometryFormat=geojson&geometryType=esriGeometryPoint&imageDisplay=1250,948,96&lang=de&layers=all:ch.swisstopo.geologie-gletschermaechtigkeit&limit=10&mapExtent=2583589.961,1082615.293,2586714.961,1084985.293&returnGeometry=true&sr=2056&tolerance=10) -> 58.290000915527344
[htmlpopup](https://mf-chsdi3.dev.bgdi.ch/ltfas_swisstopo_geologie_gletschermaechtigkeit/2103152022/rest/services/ech/MapServer/ch.swisstopo.geologie-gletschermaechtigkeit/9551/htmlPopup?coord=2585231.3574218745,1083917.0214843748&imageDisplay=1250,948,96&lang=de&mapExtent=2583589.961,1082615.293,2586714.961,1084985.293&sr=2056) -> 58.29

according to the issue in jira https://jira.swisstopo.ch/browse/BGDIDIC-1927 the volume should be delivered with two digits. Is the testlink here already showing the new data?

Reply:
I see what you mean. It's strange because the values for the volumes are already rounded to 2 digits in the .shp and ArcGIS. When importing into FME, also two digits. However when it gets exported from FME to PG, the values change and many decimals are added. I think I will add a post-import SQL rounding at the end of the FME workbench (or if there is a known reason for this behaviour and a solution to fix the problem at its roots, let me know).

Reply:
I guess it has something to do with double precision vs. numeric

Reply:
Given the finite binary representation of floats, there are many numbers that do not fit precisely a discrete step, which leads to this result upon conversion between string<>float. Either you store as a formatted string or as numeric with specified precision.

Reply:
Alright, updating the column type from double precision to numeric fixed the problem

Reply:
thank you @faselm and @b4l for solving this!

Reply:
we will have to deploy the whole database for this change.